Automattic/woocommerce-payments
# fix: Store API tokenized cart nonce verification

#### Changes proposed in this Pull Request









[Here](https://github.com/Automattic/woocommerce-payments/pull/8644#pullrequestreview-2061781370) we realized some weird behavior with the tokenized cart PRBs when a block-based theme was used.

Every `POST` Store API request was throwing a `woocommerce_rest_invalid_nonce` error code.
It turns out that the behavior is due to [this middleware](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/assets/js/middleware/store-api-nonce.js), altering each Store API requests.

Checked in here with Team Rubik: p1715950599536899-slack-C02TS23QJ1X
They suggested we could skip nonce validations when a `Cart-Token` header is present ( https://github.com/woocommerce/woocommerce/issues/42341 ).

I am keeping the nonce validation, but instead overwriting the nonce on the server-side, before the request is processed.

#### Testing instructions





- Enable the "Enable Cart-Token implementation for PRBs" flag in your dev tools (you'll need to update them)
- Ensure you have Google Pay/Apple Pay enabled in the merchant's settings
- Ensure you have a block-based theme installed on your site
- Go to a product page on your site
- Open the browser's network requests
  - You shouldn't see `403` error codes due to invalid nonces

-------------------


**Post merge**
